Mu2eG4/inc/Mu2eG4WorkerRunManager.hh

```diff
@@ -51,7 +51,7 @@ namespace mu2e {
     void initializeUserActions(const G4ThreeVector& origin_in_world);
     void initializeRun(art::Event* art_event);
     void processEvent(art::Event*);
-    G4Event* generateEvt(G4int i_event);
+    G4Event* generateEvt(std::string i_event, std::string i_subrun, std::string i_run);
```

Comment:
The signature of generateEvt() strikes me as non-conventional.  If we pass event,run,subrun, I expect three integer arguments.  Or a const art::EventID reference.   Alternatively, if the function only needs a unique string, pass it a string.  But I see in the implementation that i_events is converted from string back into integer.  So the single string option is not what we want, but please consider using one of the first two options I suggested.

Reply:
I do need the run, subrun, and event numbers as strings in the function generateEvt in order to make the input string for the hash:
string msg = "r" + i_run + "s" + i_subrun + "e" + i_event + salt_;

However, I can use the const art::EventID reference as you suggest for the argument of the function and convert that to a string inside the function.